### PR TITLE
chore: refactor to a monorepo including core

### PR DESCRIPTION
**Pull request by sauyon**
@aarnphm @frostming I think this was somewhat overdue; I'll continue to work on the other PRs but really rebasing shouldn't be too much of a chore.

### CODE REVIEW

File: core/__pypackages__/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+*
```
Comment (aarnphm) [inline core/__pypackages__/.gitignore:1]: I don't think we should include this folder here

Is there a specific reason for this?

**Review by aarnphm — COMMENTED** [review]
tiny comment, but still need @frostming to take another look on __pypackages__ for pdm

**Comment by aarnphm** [comment]
Also we might need to regenerate pdm.lock

**Comment by sauyon** [comment]
PDM is somewhat broken; I still need to fix the CI.

File: core/__pypackages__/.gitignore (hunk 1 above)
Comment (sauyon) [reply core/__pypackages__/.gitignore:1]: Nope, that's a mistake.

**Comment by sauyon** [comment]
Hm, I suspect something is wrong, since `pdm lock -d` has so far taken 33 minutes; I'll probably go to sleep for today and look at this again tomorrow...

**Comment by sauyon** [comment]
I've pushed what I have in case you wanted to take a look @frostming

**Comment by sauyon** [comment]
Oh, this was expected, I guess, since pdm doesn't pick up the config from bentoml about our dependency overrides.

**Comment by sauyon** [comment]
@ssheng @parano not sure who controls readthedocs, I think it'll have to be updated to render the correct directory, though we could possibly just leave docs at root.

**Comment by sauyon** [comment]
Hm, I don't think PDM monorepo support is quite there yet; it probably makes more sense to just treat the two projects as separate and have CI just operate in each directory separately.

**Pull request closed without merging**